opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/NodeLinkRestService.java

```diff
@@ -0,0 +1,249 @@
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.rest.v2;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+import org.opennms.netmgt.dao.api.NodeDao;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.web.enlinkd.BridgeLinkNode;
+import org.opennms.web.enlinkd.BridgeLinkRemoteNode;
+import org.opennms.web.enlinkd.CdpElementNode;
+import org.opennms.web.enlinkd.CdpLinkNode;
+import org.opennms.web.enlinkd.EnLinkdElementFactoryInterface;
+import org.opennms.web.enlinkd.IsisElementNode;
+import org.opennms.web.enlinkd.IsisLinkNode;
+import org.opennms.web.enlinkd.LldpElementNode;
+import org.opennms.web.enlinkd.LldpLinkNode;
+import org.opennms.web.enlinkd.OspfElementNode;
+import org.opennms.web.enlinkd.OspfLinkNode;
+import org.opennms.web.rest.model.v2.CdpElementNodeDTO;
+import org.opennms.web.rest.v2.api.NodeLinkRestApi;
+import org.opennms.web.rest.model.v2.BridgeLinkNodeDTO;
+import org.opennms.web.rest.model.v2.BridgeLinkRemoteNodeDTO;
+import org.opennms.web.rest.model.v2.CdpLinkNodeDTO;
+import org.opennms.web.rest.model.v2.EnlinkdDTO;
+import org.opennms.web.rest.model.v2.IsisElementNodeDTO;
+import org.opennms.web.rest.model.v2.IsisLinkNodeDTO;
+import org.opennms.web.rest.model.v2.LldpElementNodeDTO;
+import org.opennms.web.rest.model.v2.LldpLinkNodeDTO;
+import org.opennms.web.rest.model.v2.OspfElementNodeDTO;
+import org.opennms.web.rest.model.v2.OspfLinkNodeDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+@Component
+@Transactional
+public class NodeLinkRestService implements NodeLinkRestApi {
+
+    @Autowired
+    private EnLinkdElementFactoryInterface enLinkdElementFactory;
+
+    @Autowired
+    private NodeDao m_nodeDao;
+
+    @Override
+    public EnlinkdDTO getEnlinkd(String nodeId) {
+        return new EnlinkdDTO()
+                .withLldpLinkNodeDTOs(getLldpLinks(nodeId))
+                .withBridgeLinkNodeDTOS(getBridgelinks(nodeId))
```

Comment:
Need more error protection. For ex : `getLldpLinks` can throw exception but that shouldn't stop population other elements

Reply:
I probably am going to disagree with this one. If there is any unhandled exception during the rest call, then that shouldn't happen, we should fix it, but not skip it. If a user wants to check individual links, there are rest apis for individual links.  The real problem here is there might have duplicate DB visits here which may be fine for now but could be harmful when we have things on the web. I've refactored the code.